treescript/src/treescript/github/versionmanip.py

```diff
@@ -108,83 +93,43 @@ async def do_bump_version(repo_path, files, next_version, source_repo):
+        branch (str): The branch to bump the files on.
 
     Raises:
         TaskverificationError: if a file specified is not allowed, or
                                if the file is not in the target repository.
 
     Returns:
-        int: the number of commits created.
+        Dict[str, str]: An object mapping file name to new contents.
 
     """
-    changed = False
-    saved_next_version = next_version
-
+    changes = {}
+    file_contents = await client.get_files(files, branch=branch)
     for file_ in files:
-        abs_file = os.path.join(repo_path, file_)
         if file_ not in ALLOWED_BUMP_FILES:
             raise TaskVerificationError("{} is not in version bump whitelist".format(file_))
-        if not os.path.exists(abs_file):
-            raise TaskVerificationError("{} is not in repo".format(abs_file))
-
-        VersionClass = _find_what_version_parser_to_use(file_, source_repo)
-        curr_version = get_version(file_, repo_path, source_repo)
-        next_version = VersionClass.parse(saved_next_version)
-
-        try:
-            is_esr = curr_version.is_esr
-        except AttributeError:  # Fenix does not expose the is_esr attribute
-            is_esr = False
-
-        # XXX In the case of ESR, some files (like version.txt) show version numbers without `esr`
-        # at the end. next_version is usually provided without `esr` too.
-        # That's why we do this late minute replacement and why we reset `next_version` at every
-        # cycle of the loop
-        if is_esr and not any(
-            (
-                next_version.is_esr,  # No need to append esr again
-                # We don't want XX.Ya1esr nor XX.YbNesr
-                next_version.is_aurora_or_devedition,
-                next_version.is_beta,
-            )
-        ):
-            next_version = VersionClass.parse("{}esr".format(next_version))
-
-        if next_version < curr_version:
-            log.warning("Version bumping skipped due to conflicting values: " "(next version {} is < current version {})".format(next_version, curr_version))
+
+        contents = file_contents[file_]
+
+        VersionClass = _find_what_version_parser_to_use(file_, client.repo)
+        curver = get_version(contents, VersionClass)
+        nextver = VersionClass.parse(next_version)
+
+        if nextver < curver:
+            log.warning("Version bumping skipped due to conflicting values: (next version {} is < current version {})".format(nextver, curver))
             continue
-        elif next_version == curr_version:
+        elif nextver == curver:
             log.info("Version bumping skipped due to unchanged values")
             continue
         else:
-            changed = True
-            replace_ver_in_file(abs_file, curr_version, next_version)
-
-    return changed
+            new_contents = contents.replace(str(curver), str(nextver))
+            if contents == new_contents:
+                raise Exception("File was not changed!")
```

Comment:
Nit: this way don't have to catch a broad `Exception`

```suggestion
                raise TreeScriptError("File was not changed!")
```